### PR TITLE
WT-10614 Coverity analysis defect 135089: Dereference before null check

### PR DESCRIPTION
**Pull request by amiyuki7**
page->modify is valid

### CODE REVIEW

**Review by etienneptl — APPROVED** [review]
LGTM!

File: src/btree/row_modify.c [hunk 1]
```diff
@@ -435,7 +435,8 @@ __wt_update_obsolete_check(
          * moved forwards. This function is used to trim update lists independently of the page
          * state, ensure there is a modify structure.
```
Comment (etienneptl) [inline src/btree/row_modify.c:436]: @amiyuki7, please can you remove the last part of the sentence: `, ensure there is a modify structure`? We already checked earlier line 354 with the `WT_PAGE_TRYLOCK` macro.

**Comment by etienneptl** [comment]
Sorry, @amiyuki7, I LGTM'ed too quickly. After thinking about it a bit more with @korteland, I have updated my comment [here](https://github.com/wiredtiger/wiredtiger/pull/9560#discussion_r1305021352). Thanks for taking care of this ticket!

File: src/btree/row_modify.c (hunk 1 above)
Comment (amiyuki7) [reply src/btree/row_modify.c]: Yep, semantic fix has been pushed.

**Review by etienneptl — APPROVED** [review]
LGTM!

**Review by korteland — APPROVED** [review]
LGTM!

**Pull request merged**